test/bats/azure-test.bats

```diff
@@ -279,3 +279,84 @@ SLEEP_TIME=1
     result=$(kubectl get pod mutate-demo --namespace default -o json | jq -r ".spec.containers[0].image" | grep @sha)
     assert_mutate_success
 }
+
+@test "validate image signed by leaf cert" {
+    teardown() {
+        wait_for_process ${WAIT_TIME} ${SLEEP_TIME} 'kubectl delete certificatestores.config.ratify.deislabs.io/certstore-inline --namespace default --ignore-not-found=true'
+        wait_for_process ${WAIT_TIME} ${SLEEP_TIME} 'kubectl delete pod demo-leaf --namespace default --force --ignore-not-found=true'
+        wait_for_process ${WAIT_TIME} ${SLEEP_TIME} 'kubectl delete pod demo-leaf2 --namespace default --force --ignore-not-found=true'
+
+        # restore the original notary verifier for other tests
+        wait_for_process ${WAIT_TIME} ${SLEEP_TIME} 'kubectl apply -f ./config/samples/config_v1beta1_verifier_notary.yaml'
+    }
+
+    # configure the default template/constraint
+    run kubectl apply -f ./library/default/template.yaml
+    assert_success
+    run kubectl apply -f ./library/default/samples/constraint.yaml
+    assert_success
+
+    # add the root certificate as an inline certificate store
+    cat <<EOF | kubectl apply -f -
+apiVersion: config.ratify.deislabs.io/v1beta1
+kind: CertificateStore
+metadata:
+  name: certstore-inline
+spec:
+  provider: inline
+  parameters:
+    value: |
+$(cat ~/.config/notation/truststore/x509/ca/leaf-test/root.crt | sed 's/^/      /g')
+EOF
+
+    # configure the notary verifier to use the inline certificate store
+    cat <<EOF | kubectl apply -f -
```

Comment:
curious what your thoughts are on making this a standalone rather than redefining every time in the test body?

Reply:
moved to yaml files under test directory.